portfolio/src/main/webapp/tasksApi.js

```diff
@@ -0,0 +1,203 @@
+var GoogleAuth;
+
+// Scope for read access to Tasks API
+const SCOPE_TASKS_READ = 'https://www.googleapis.com/auth/tasks.readonly';
+
+const TASKS_CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+const TASKS_DISCOVERY_DOCS =
+    'https://www.googleapis.com/discovery/v1/apis/tasks/v1/rest';
+
+var TASKS_API_KEY = '';
+
+function fetchApiKey() {
+  fetch('./appConfigServlet')
+      .then(response => response.json())
```

Comment:
I think this chaining is unnecessary. You can compute response.json() in a line before TASKS_API_KEY = ...

Reply:
Do you mean like ```responseJson = response.json();```?